plugins/scaffolder-node/report-alpha.api.md

```diff
@@ -23,6 +23,7 @@ export type AutocompleteHandler = ({
 }) => Promise<{
   results: {
     title: string;
+    id?: string;
```

Comment:
I think that we should make this `id` required, and name be optional.

Name would be the human readable version that's rendered in the UI if it's available, otherwise it falls back to the `id` instead.

it's technically a breaking change, but because this stuff is on alpha, it's fine for us to release this as a `minor` change.

I see that you've already updated the bitbucket one too so that's great, but I think it makes sense to make that change now.

Reply:
I think we would need to define what the ID holds then. E.g. for [this structure](https://github.com/backstage/backstage/pull/27367/files#diff-a3a6fd3b5458cb6adb4e14446e4fbae8be3226c7dc7dadbd64e66abd957eef99R100) there is no `uuid`. Also, the gitlab `publish:mr` [action](https://github.com/backstage/backstage/blob/a096e9ff2d553c7ead7df62794b33babff1d3f37/plugins/scaffolder-backend-module-gitlab/src/actions/gitlabMergeRequest.ts) requires `group` and `title` and not the projects id.

Therefore, I do not know what the whole impact of that change would be.

Reply:
Hey @benjdlambert, do you really want to make this change now? Should the dependant modules be updated right away? How would you want to go about this?

Reply:
Hey @swnia, sorry for the delay in getting back, had a week off last week and feeling pretty sick the beginning of this week.

I think that we should make this change. I think that it was a bit of an oversight at the beginning not to have `id` and `title?` in the first place, and the title is what is rendered in the dropdown but the ID is what is used to call the value unless it's special.

I think it's OK to extend if needed, but that should be on a per autocomplete basis, and perhaps cast in the usage of the `.autocomplete()` call in the field extensions themselves.

`scaffolderClient.autocomplete<GitlabProjects>()` or something?

Reply:
The RepoUrlPicker does not seem to be using the `.autocomplete()` call... I'll see what changes are necessary to make it work for Gitlab.

Reply:
It's used in the `GitlabRepoUrlPicker` though right? It would be there that we could cast if necessary.

Reply:
Yes, my search over the code was too narrow which 🤦 So, yes, it's [here](https://github.com/backstage/backstage/pull/27367/files#diff-8c91e7f5b7e1bfd5878797e0bc07aab845b75c00a0df03caee9b0f2b68b69548R86)